---
title: "CTDPathSim"
author: "Banabithi Bose, Matt Moravec,  Serdar Bozdag"
date: "5/12/2020"
output: rmarkdown::html_vignette
r_packages: rmarkdown
vignette: > 
    %\VignetteIndexEntry{mirDriver} 
    %\VignetteEngine{knitr} 
    %\VignetteEncoding[UTF-8]{inputenc}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Table of Contents
    * [Introduction](#Introduction)
    * [Example Data Sets](#Example-Data-Sets)
        * [Step 1 Data Sets](#Step-1-Data-Sets)
            * [RunDeconvMethyl Data](#RunDeconvMethyl-Data)
            * [PlotDeconvMehtyl Data](#PlotDeconvMehtyl-Data)
            * [ProbeTogene450Fun Data](#ProbeTogene450Fun-Data)
            * [ProbeTogene27Fun Data](#ProbeTogene27Fun-Data)
            * [sampleMethylFun Data](#sampleMethylFun-Data)
        * [Step 2 Data Sets](#Step-2-Data-Sets)
            * [RunDeconvExpr Data](#RunDeconvExpr-Data)
            * [sampleExprFun Data](#sampleExprFun-Data)
        * [Step 3 Data Sets](#Step-3-Data-Sets)
            * [GetSampleDE Data](#GetSampleDE-Data)
            * [GetSampleDM Data](#GetSampleDM-Data)
            * [GetPath Data](#GetPath-Data)
        * [Step 4 Data Sets](#Step-4-Data-Sets)
            * [FindSim Data](#FindSim-Data)
    * [Using the Functions](#Using-the-Functions)
        * [RunDeconvMethyl](#RunDeconvMethyl)
        * [PlotDeconvMehtyl](#PlotDeconvMehtyl)
        * [ProbeTogene450Fun](#ProbeTogene450Fun)
        * [ProbeTogene27Fun](#ProbeTogene27Fun)
        * [sampleMethylFun](#sampleMethylFun)   
        * [RunDeconvExpr](#RunDeconvExpr)
        * [sampleExprFun](#sampleExprFun)
        * [GetSampleDE](#GetSampleDE)
        * [GetSampleDM](#GetSampleDM)
        * [GetPath](#GetPath)
        * [FindSim](#FindSim)
    * [Example Workflow](#Combined-Example-Workflow)
    * [Summary](#Summary)
    * [Acknowledgements](#Acknowledgements)


## <a id = "Introduction"></a> Introduction
In the context of cancer research and drug development tumor derived cell lines are used to determine the response by the patient to the drug. However, the drug response to the cell line do not always accurately represent the patient’s response to the treatment. To bridge the gap between patient i.e. tumor sample-based and cell-line-based drug response studies, B. Bose et. al, 2020, proposed a computational pipeline called CTDPathSim that measures the biological pathway activity-based similarity between a tumor sample and cell lines using integrative multi-omics molecular profiles. CTDPathSim also accounts for the heterogeneity of bulk tumor profiles by deconvolving the effect of various cell types. The current project is to develop CTDPathSim tool in the form of a Bioconductor R package to be utilized by the scientific community. 

```{r}
library(CTDPathSim)
```


## <a id = "Example-Data-Sets"> </a> Example Data Sets
These shows a small view of all data sets used in each step. 


## <a id = "Step-1-Data-Sets"> </a> Step 1 Data Sets
These are datasets used in the step one of the pipeline.

## <a id = "RunDeconvMethyl-Data"> </a> RunDeconvMethyl Data
RunDeconvMethylData() is the first function in step 1 of CTDPathSim R package. This function takes two arguments, “Example.Methylation” and “markers.ovr”. The argument “Example.Methylation” should be a R dataframe object where the columns are patient barcodes and rows are probes. The values that align with each probe and patient are the methylation values. In the CTDPathSim package an example of a data frame like this is included in the package Example.Methylation.450k.50.rda.The second argument, markers_ovr, is a vector
of the probes. These are values in cg notation in the data frame above. This R dataframe object holds roughly 500 probes. The output of this function is a list of characteristic about the patient. The methylation and proportion sections of this list are used to make two data sets that are used later in the pipeline.

## <a id = "Example.Methylation"> </a> Example.Methylation
```{r}
load(system.file("extdata", "Example.Methylation.450k.50.rda", package="CTDPathSim"))
print(Example.Methylation.450K[1:3,1:2])
```
## <a id = "markers_ovr"> </a> markers_ovr
This datasets is a vector of all the probes used in the study.
```{r}
load(system.file("extdata", "markers_ovr_450.rda", package = "CTDPathSim"))
print(markers_ovr[1:5])
```

## <a id = "PlotDeconvMethyl-Data"> </a> PlotDeconvMethyl Data
This will overview the datasets used in the function and the datasets output by the function. Three of the arguments and data input are the input to the RunDeconvMethyl() function and the output of the RunDeconvMethyl() function and are described previously. The other inputs are the reference methylation file and the cn file. The reference methylation file is a data frame where the columns are genes and the rows are probes. The corresponding values are the methylation values. The cn file is a R dataframe object that has the class name for all the genes. Another input for this function is the markers file. The file name is markes_ovr and has about 500 markers in it.The output of this function is a pdf correlation plot that compares the deconvoluted methylation profile to the reference methylation profile.

## <a id = "markers_ovr"> </a> markers_ovr
This data set is described and viewable in the RunDeconvMethyl() function

## <a id = "Example.Methylation"> </a> Example.Methylation
This data set is described and viewable in the RunDeconvMethyl() function

## <a id = "complete_ref_meth"> </a> complete_ref_meth
This data set is the reference methylation profile.

```{r}
load(system.file("extdata", "complete_ref_meth450_50.rda", package = "CTDPathSim"))
print(complete_ref_meth450[1:3,1:5])
```

## <a id = "cn"> </a> cn
This file is a R dataframe object with the genes and their class
```{r}
cn<-xlsx::read.xlsx2(file = system.file("extdata", "ref_names.xlsx", package = "CTDPathSim"),header=TRUE,sheetIndex = 1)
print(cn[1:5,])
```


## <a id = "ProbeTogene450Fun-Data"> </a> ProbeTogene450Fun Data
This section will cover the input and output data for the third function of the first step. One of the two input data values is the output of the first function and a portion is viewable above. The other is a data frame that contains the reference gene and group of each probe.

## <a id = "A_450"> </a> A_450
This data set is a R dataframe object that includes the probes, the symbol and synonyms
```{r}
A_450<-xlsx::read.xlsx(system.file("extdata","Methylation.Probe.Annotation450.50.xlsx",package="CTDPathSim"), header=TRUE,sheetIndex = 1)
print(A_450[1:3,1:3])
```

## <a id = "Deconv_Methylation"> </a> Deconv_Methylation
This data set is a R dataframe with methylation values for each probe and cell type
```{r}
load(system.file("extdata", "Example.Deconv.Methylation.450k.50.rda", package = "CTDPathSim"))
print(Deconv_methylation[1:3,1:3])
```


## <a id = "ProbeTogene27Fun-Data"> </a> ProbeTogene27Fun Data
This function uses similar inputs and makes a similar output but used 27k probes instead of 450k probes.

## <a id = "sampleMethylFun-Data"> </a> sampleMethylFun Data
This section will cover the data inputs and output of the fifth and final function of step one. The two inputs to this function are the deconvoluted gene specific methylation R dataframe object and the deconvoluted proportion R dataframe object. The gene specific methylation R dataframe object is viewable above. The proportion data frame has a proportion value for each cell type for each patient (See table 1).This function has output of a R dataframe object for the methylation value for each patient that is stored in a directory labeled Sample_methylation.

## <a id = "Deconv_meth_gene"> </a> Deconv_meth_gene
This data set is a R dataframe object with the deconvoluted methylation values for each gene and cell type.
```{r}
load(system.file("extdata", "Example.Deconv.meth.gene.450k.rda", package = "CTDPathSim"))
print(Deconv_meth_gene[1:3,1:3])
```

## <a id = "Deconv_proportions"> </a> Deconv_proportions
This dataset is a R dataframe object with the proportions of each cell type for each patient.
```{r}
load(system.file("extdata", "Example.Deconv.proportions.450K.rda", package = "CTDPathSim"))
print(Deconv_proportions[1:3, 1:3])
```

## <a id = "Output"> </a> Output
The output of this function is a list.
```{r}
load(system.file("extdata", "stage1_result_ct.rda", package = "CTDPathSim"))
print(stage1_result_ct)
```



## <a id = "Step-2-Data-Sets"> </a> Step 2 Data Sets
These are datasets used in the step two of the pipeline.

## <a id = "RunDeconvExpr-Data"> </a> RunDeconvExpr Data
This will cover the input and output datasets for the first function of step two. One of the inputs is the deconvoluted proportions R dataframe object. This R dataframe object is viewable
above. The other input is expression data. The columns are patients, the rows are genes and the values corresponding to a specific patient and gene are the expression values.

## <a id = "RnaSeq_fpkm_hgnc_data"> </a> RnaSeq_fpkm_hgnc_data
This data set is a R dataframe object with the expression values for each probe and cell type.
```{r}
load(system.file("extdata", "Example.Expression.50.rda", package = "CTDPathSim"))
print(RnaSeq_fpkm_hgnc_data[1:3,1:3])
```


## <a id = "Deconv_proportions"> </a> Deconv_proportions
This dataset is described above.


## <a id = "sampleExprFun-Data"> </a> sampleExprFun Data
The input datasets for the second function of step two is the deconvoluted expression and deconvoluted proportion datasets, both of which are viewable above. The output is a collection of expression profile R dataframe objects for each patient stored in the directory labeled Sample_expression.

## <a id = "Deconv_expression"> </a> Deconv_expression
This data sets is a R dataframe object with an expression value for each gene and cell type.
```{r}
load(system.file("extdata", "Example.Deconv.expression.rda", package = "CTDPathSim"))
print(Deconv_expression[1:3,1:3])
```
## <a id = "Deconv_proportions"> </a> Deconv_proportions
This dataset is described above.

## <a id = "Step-3-Data-Sets"> </a> Step 3 Data Sets
These are datasets used in the step three of the pipeline.

## <a id = "GetSampleDE-Data"> </a> GetSampleDE Data
The input datasets for the first function of the third step is the expression profile R dataframe object and is visible above. The output is a series of R dataframe objectsthat hold the DE genes for each patient. 

## <a id = "RnaSeq_fpkm_hgnc_data"> </a> RnaSeq_fpkm_hgnc_data
This dataset is described above.

## <a id = "GetSampleDM-Data"> </a> GetSampleDM Data
The input datasets for the first function of the third step is the gene methylation profile R
dataframe object and is visible above. It has patients for the columns and genes for the rows
and the values that correspond to one patient and gene are the methylation values.

## <a id = "All_methylation"> </a> All_methylation
This data set is a R dataframe object with the methylation value for each gene and patient.
```{r}
load(system.file("extdata", "Example.gene.centric.methylation.50.rda", package = "CTDPathSim"))
print(All_methylation[20:23,1:3])
```

## <a id = "GetPath-Data"> </a> GetPath Data
The inputs for the third and final function of step three are a data frame with all the cancer genes and a data frame with the DE genes for one 
patient. 
## <a id = "CancerGeneList"> </a> CancerGeneList
This data set is a R dataframe object with values of frequently mutated cancer driver genes.
```{r}
load(system.file("extdata", "Example.CancerGeneList.rda", package = "CTDPathSim"))
print(CancerGeneList[1:5,])
```
## <a id = "pat_de_genes"> </a> pat_de_genes
This data set is a R dataframe object with the patients DE genes.
```{r}
load(system.file("extdata", "Example.patient1.de.genes.rda", package = "CTDPathSim"))
print(pat_de_genes[1:5,])
```


## <a id = "Step-4-Data-Sets"> </a> Step 4 Data Sets
These are datasets used in the step four of the pipeline.

## <a id = "FindSim-Data"> </a> FindSim Data
The function for the final step, step 4, has six input data sets: data frame for patient DE genes, data frame for patient pathway, data frame for patient expression, data frame for cell line DE genes, data frame for cell line pathway and data frame for cell line expression. 

## <a id = "pat_de_genes"> </a> pat_de_genes
This dataset is described above.

## <a id = "pat_reactome"> </a> pat_reactome
This data set is a R dataframe object with the patients reactome pathways.
```{r}
load(system.file("extdata", "Example.patient1.pathways.rda", package = "CTDPathSim"))
print(pat_reactome[1:3,])
```

## <a id = "expression"> </a> expression
This data set is a R dataframe object with the patients reactome pathways.
```{r}
load(system.file("extdata", "Example.patient1.deconvoluted.expression.rda", package = "CTDPathSim"))
print(expression[1:3,])
```

## <a id = "cell_de_genes"> </a> cell_de_genes
This data set that looks similar to the patient DE genes data frame.

## <a id = "cell_reactome"> </a> cell_reactome
This data set that looks similar to the patient reactome data frame.

## <a id = "ccle_expr"> </a> ccle_expr
This data set that looks similar to the patient expression data frame.

## <a id = "Using-the-Functions"> </a> Using the Functions
This section will take the user through the functions in the package. It will include what is does and how to use it.



## <a id = "RunDeconvMethyl"> </a> RunDeconvMethyl
RunDeconvMethyl(), computes the deconvolution methylation profile. This profile includes deconvoluted methylation values for each probe location.

Example.Methylation - A Dataframe for the example methylation values for the patients and the probes.

markers_ovr - A vector of the probe markers.

```{r}
##RUNNING THE FUNCTION WITH EXAMPLE DATASETS
#load(system.file("extdata", "Example.Methylation.450k.rda", package = "CTDPathSim"))
#load(system.file("extdata", "markers_ovr_450.rda", package = "CTDPathSim"))
#stage1_result_ct<-RunDeconvMethyl(Example.Methylation=Example.Methylation.450K, markers_ovr=markers_ovr)
```


## <a id = "PlotDeconvMethyl"> </a> PlotDeconvMethyl
PlotDeconvMethyl(), creates a correlation plot to show the correlation between the cell types used for deconvolution. User can use this plot to choose the appropriate labels for the cell types for the deconvoluted methylation profiles.

Example.Methylation - A R dataframe object with the expression values for each cell type

markers_ovr - A R dataframe object that includes all the probes used in the study

complete_ref_meth - A R dataframe object with the reference methylation values.

stage1_result_ct - A list that contains the otuput of runDeconvMethyl

cn - A R dataframe object with the reference cell types and their class

```{r}
#load(system.file("extdata", "Example.Methylation.450k.50.rda", package = "CTDPathSim"))
#load(system.file("extdata", "complete_ref_meth450.rda", package = "CTDPathSim"))
#load(system.file("extdata", "markers_ovr_450.rda", package = "CTDPathSim"))
#load("~/CTDPathSim/inst/extdata/old_stage1_result_ct_450K.rda")
#load(system.file("extdata", "stage1_result_ct_450K.rda", package = "CTDPathSim"))
#library(xlsx)
# Create a vector of colors representing the class of each reference
#cn<-xlsx::read.xlsx(file = "~/CTDPathSim/inst/extdata/ref_names.xlsx",header=TRUE,sheetIndex = 1)
#pdf("~/CTDPathSim/inst/extdata/_cor_matrix_stage1_450K.pdf")
#print(PlotDeconvMethyl(Example.Methylation=Example.Methylation.450K,markers_ovr=markers_ovr,
                    #complete_ref_meth=complete_ref_meth450,stage1_result_ct=stage1_result_ct,cn=cn))
#dev.off()
```


## <a id = "ProbeTogene450Fun"> </a> ProbeTogene450Fun
ProbeTogene450Fun(), convertsthe methylation data from the probes to the genes for the 450k probes.

A_450 - A R dataframe object that includes the probes, the symbol and synonyms

Deconv_methylation - A R dataframe with methylation values for each probe and cell type

```{r}
#load(system.file("extdata", "Methylation.Probe.Annotation450.rda", package = "CTDPathSim"))
#load(system.file("extdata", "Example.Deconv.Methylation.450k.50.rda", package = "CTDPathSim"))
#Deconv_meth_gene<-ProbeTogene450Fun(A_450=A_450,Deconv_methylation=Deconv_methylation)
```


## <a id = "ProbeTogene27Fun"> </a> ProbeTogene27Fun
ProbeTogene27Fun(), also converts the methylation data from probes to genes but does so with 27k probes.

A_27 - A R dataframe object that includes the probes, the symbol and synonyms

Deconv_methylation - A R dataframe with methylation values for each probe and cell type

```{r}
#Deconv_meth_gene<-ProbeTogene27Fun(A_27,Deconv_methylation)
```

## <a id = "sampleMethylFun"> </a> sampleMethylFun
This function makes a sample specific methylation file for each sample and saves it in a directory entitled, Sample methylation.

Deconv_meth_gene - A R dataframe object with the deconvoluted methylation values for each gene and cell type

Deconv_proportions - A R dataframe object with the proportions of each cell type for each patient

```{r}
##RUNNING THE FUNCTION WITH EXAMPLE DATASETS
load(system.file("extdata", "Example.Deconv.Methylation.450k.50.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.Deconv.proportions.450K.rda", package = "CTDPathSim"))

##SAMPLE SPECIFIC METHYLATION FILES ARE SAVED IN Sample_methylation FOLDER
#sampleMethylFun(Deconv_meth_gene,Deconv_proportions)
```


## <a id = "RunDeconvExpr"> </a> RunDeconvExpr
RunDeconvExpr(), creates a deconvoluted expression profile.

RnaSeq_fpkm_hgnc_data - A R dataframe object with the expression values for each probe and cell type


Deconv_proportions - A R dataframe object with the proportion of each cell type for each patient

```{r}
load(system.file("extdata", "Example.Expression.50.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.Deconv.proportions.450K.rda", package = "CTDPathSim"))
#Deconv_expression<-RunDeconvExpr(RnaSeq_fpkm_hgnc_data,Deconv_proportions)
#save(Deconv_expression,file="~/Example.Deconv.expression.rda")
```

## <a id = "sampleExprFun"> </a> sampleExprFun
sampleExprFun(), is designed to create an expression file for each sample.

Deconv_expression - A R dataframe object with an expression value for each gene and cell type

Deconv_proportions - A R dataframe object with proportions of each cell type for each patient

```{r}
##RUNNING THE FUNCTION WITH EXAMPLE DATASETS
load(system.file("extdata", "Example.Deconv.expression.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.Deconv.proportions.450K.rda", package = "CTDPathSim"))
##SAMPLE SPECIFIC EXPRESSION FILES ARE SAVED IN Sample_expression FOLDER
#sampleExprFun(Deconv_expression,Deconv_proportions)
```

## <a id = "GetSampleDE"> </a> GetSampleDE
GetSampleDE(), is designed to find the median expression of a gene for a patient. The result is stored in a directory named Patient_DE_genes as an individual data frame for each patient.

RnaSeq_fpkm_hgnc_data - R dataframe object with the expression values for each probe and cell type

parallel - A Boolean value for whether or not the user wants to run this function in multi-core. If the user is on windows parallel must be FALSE

ncores - An integer value for the number of cores the use if the code is run in parallel

```{r}
##RUNNING THE FUNCTION
load(system.file("extdata", "Example.Expression.50.rda", package = "CTDPathSim"))
<<<<<<< HEAD
#GetSampleDE(RnaSeq_fpkm_hgnc_data=RnaSeq_fpkm_hgnc_data,parallel=FALSE,ncores=5)
=======
GetSampleDE(RnaSeq_fpkm_hgnc_data=RnaSeq_fpkm_hgnc_data,parallel=FALSE,ncores=5)
>>>>>>> 14209712fa443e93cbd50dd499c4d186bf5a6d35
```

## <a id = "GetSampleDM"> </a> GetSampleDM
GetSampleDM(), finds the median methylation value between cell types for each patient or cell line. The resulting data frames, one per sample, are then stored in a directory labeled Patient_DM_Genes.

All_methylation - A R dataframe object with the methylation value for each gene and patient

parallel - A Boolean value for whether or not the user wants to run this function in multi-core. If the user is on windows parallel must be FALSE

ncores - An integer value for the number of cores the use if the code is run in parallel

```{r}
load(system.file("extdata", "Example.gene.centric.methylation.50.rda", package = "CTDPathSim"))
<<<<<<< HEAD
#GetSampleDM(All_methylation=All_methylation,parallel=FALSE,ncores=5)
=======
GetSampleDM(All_methylation=All_methylation,parallel=FALSE,ncores=5)
>>>>>>> 14209712fa443e93cbd50dd499c4d186bf5a6d35
```

## <a id = "GetPath"> </a> GetPath
GetPath(), finds the pathway between cancer genes and specific patient DE genes.

CancerGeneList - A R dataframe object with values of frequently mutated cancer driver genes

pat_de_genes - A R dataframe object with the patients DE genes

```{r}
load(system.file("extdata", "Example.CancerGeneList.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.patient1.de.genes.rda", package = "CTDPathSim"))
<<<<<<< HEAD
#x<-GetPath(CancerGeneList,pat_de_genes)
=======
x<-GetPath(CancerGeneList,pat_de_genes)
>>>>>>> 14209712fa443e93cbd50dd499c4d186bf5a6d35
```

## <a id = "FindSim"> </a> FindSim
This function uses the DE genes, DM genes and pathway of both the patient sample and cell line as components for the Spearman correlation and returns a R dataframe object that holds the correlation values for the patients and cell lines. 

pat_de_genes - A R dataframe object with all of the patients DE genes.

pat_reactome - A R dataframe object with the patients reactome pathways.

expression - A R dataframe object with the patients expression values for each gene and cell type.

cell_de_genes - A R dataframe object with all of the cell line's DE genes.

cell_reactome - A R dataframe object with the cell line's reactome pathways.

ccle_expr - A R dataframe object with the cell line's expression values for each gene and cell type.

```{r}
load(system.file("extdata", "Example.patient1.de.genes.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.patient1.pathways.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.patient1.deconvoluted.expression.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.cellLine1.de.genes.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.cellLine1.pathways.rda", package = "CTDPathSim"))
load(system.file("extdata", "Example.cellLine1.expression.rda", package = "CTDPathSim"))
patient1.cellLine1.spearman<-FindSim(pat_de_genes,pat_reactome,expression,
cell_de_genes,cell_reactome,ccle_expr)
```


## <a id = "Example-Workflow"> </a> Example Workflow
The first three steps must be done for both the patient and cell line so the final step can compute the similarity score between the patient and cell line. This code will not be runnable as all code is run previously in the vignette.

##Step 1
load("~/CTDPathSim/inst/extdata/Example.Methylation.450k.50.rda")
load("~/CTDPathSim/inst/extdata/markers_ovr_450.rda")
stage1_result_ct<-RunDeconvMethyl(Example.Methylation=Example.Methylation.450K, markers_ovr=markers_ovr)
Deconv_methylation<-stage1_result_ct$methylation
Deconv_proportions <-stage1_result_ct$proportions
#Plotting the data is not required to complete the pipeline
#The one probe to gene function will be used depending on the probes used. In this case, the 450k probes are used
load("~/CTDPathSim/inst/extdata/Methylation.Probe.Annotation450.rda")
Deconv_meth_gene<-ProbeTogene450Fun(A_450=A_450,Deconv_methylation=Deconv_methylation)
sampleMethylFun(Deconv_meth_gene,Deconv_proportions)
load("~/CTDPathSim/inst/extdata/Example.Expression.50.rda")
Deconv_expression<-RunDeconvExpr(RnaSeq_fpkm_hgnc_data,Deconv_proportions)
save(Deconv_expression,file="~/Example.Deconv.expression.rda")
sampleExprFun(Deconv_expression,Deconv_proportions)
GetSampleDE(RnaSeq_fpkm_hgnc_data=RnaSeq_fpkm_hgnc_data,parallel=FALSE,ncores=5)
load("~/CTDPathSim/inst/extdata/Example.gene.centric.methylation.50.rda")
GetSampleDM(All_methylation=All_methylation,parallel=FALSE,ncores=5)
load("~/CTDPathSim/inst/extdata/Example.CancerGeneList.rda")
load("~/CTDPathSim/inst/extdata/Example.patient1.de.genes.rda")
x<-GetPath(CancerGeneList,pat_de_genes)

##Final Step does not need to be run twice as it uses data created from the code above for both the patient and cell line
load("~/CTDPathSim/inst/extdata/Example.patient1.de.genes.rda")
load("~/CTDPathSim/inst/extdata/Example.patient1.pathways.rda")
load("~/CTDPathSim/inst/extdata/Example.patient1.deconvoluted.expression.rda")
load("~/CTDPathSim/inst/extdata/Example.cellLine1.de.genes.rda")
load("~/CTDPathSim/inst/extdata/Example.cellLine1.pathways.rda")
load("~/CTDPathSim/inst/extdata/Example.cellLine1.expression.rda")
patient1.cellLine1.spearman<-FindSim(pat_de_genes,pat_reactome,expression,
cell_de_genes,cell_reactome,ccle_expr)

## <a id = "Summary"> </a> Summary
This is a four step process designed to create a similarity network between the patient sample and the patient cell line. The steps are: finding a deconvolted methylation profile, creating a deconvoluted expression profile, computing the differential genes and computing similarity. This steps are designed to be used together in as pipeline, but can also be used individually.

## <a id = "Acknowledgments"> </a> Acknowledgments
1.This material is based upon work supported by the National Science Foundation under Grant #1950826
2.This work was supported by the National Institute of General Medical Sciences of the National Institutes of Health under Award Number R35GM133657


